File: 2. python/1. basics/lecture 2/p11.py
class Solution:
    def __init__(self, n): #constructor
        for i in range(n):
            for j in range(i + 1):
                if (i + j) % 2 == 0:
                    print("1", end=" ")
                else:
                    print("0", end=" ")
            print()

''' pattern output:

1
0 1
1 0 1
0 1 0 1

'''

if __name__ == "__main__":
    size = int(input("enter size: "))

    print("11. output:")
    Solution(size)

File: 1. cpp/1. basics/lecture 2/p11.cpp

#include <iostream>
using namespace std;
class solution {
public:
    solution(int n) {
        for (int i = 0; i < n; i++) {
            for (int j = 0; j <= i; j++) {
                if ((i + j) % 2 == 0) {
                    cout << " 1 ";
                } else {
                    cout << " 0 ";
                }
            }
            cout << endl;
        }
    }
};

/* pattern output:

1
0 1
1 0 1
0 1 0 1

*/

int main() {
  int size;
  cout<<"Enter size of pattern:"<<endl;
  cin>>size;

  cout<<"11. output:"<<endl;
  solution pattern(size);

}